mrmustard/lab_dev/states/dm.py

```diff
@@ -130,22 +130,26 @@ def from_fock(
     @classmethod
     def from_ansatz(
         cls,
-        modes: Sequence[int],
+        modes: Iterable[int],
```

Comment:
Shouldn't it be Sequence? Based on [this stack overflow post](https://stackoverflow.com/questions/72157296/what-is-the-difference-between-type-hinting-a-variable-as-an-iterable-versus-a-s) a sequence is "any object that defines `__getitem__` and `__len__`" and we directly depend on `len` below.

Reply:
I thought iterable was more appropriate given that modes are assumed to be sorted. E.g. iterables like a set don't necessarily have an order.

Reply:
Anyway, i don't feel strongly about it. It can be a sequence.

Reply:
Actually I think `Collection` is the most appropriate. You're right `set` means it's not necessarily a `Sequence` as it does not implement `__getitem__` but `Collection` covers all bases